packages/hoppscotch-app/components/http/RawBody.vue

```diff
@@ -111,6 +119,25 @@ useCodemirror(
   })
 )
 
+const envVariables = useReadonlyStream(aggregateEnvs$, [])
+
+const updateBody = (newValue) => {
+  rawParamsBody.value = newValue.replace(/<<\w+>>/g, (key) => {
+    const found = envVariables.value.find(
+      (envVar) => envVar.key === key.replace(/[<>]/g, "")
+    )
+    return found ? found.value : key
+  })
+}
+
+watch(envVariables, () => {
+  updateBody(rawBodyData.value)
+})
+
+watch(rawBodyData, (newValue) => {
+  updateBody(newValue)
+})
+
```

Comment:
This is totally not the right approach to implementing this in Hoppscotch.

Hoppscotch processes requests in a step-by-step process which are broadly classified into.

- Composing (The user is still building the request, did not press the Send button yet)
- Initial Validation (Deciding whether the request should even run, when the Send button is pressed)
- Pre-Request Script Execution (A Pre-Request script is executed if defined)
- Environment Injection (After pre-request script we have all the data necessary to fill the environment variable holes, we fill them)
- Execution (Actual network request is ran)

When the send request action is fired, the pre-request scripts are initially ran. [Pre-Request scripts](https://github.com/hoppscotch/hoppscotch/wiki/Pre-Request-Scripts) allow for environment variables to be created and updated dynamically, those environments are all aggregated into an ephemeral list of environmental variables as specified here https://github.com/hoppscotch/hoppscotch/blob/6d54f21c1eed4df152b95e7c79a92d9b0fe1ae09/packages/hoppscotch-app/helpers/RequestRunner.ts#L49-L53

Your code tries to resolve environment variables during the composing phase itself, which doesn't take into account the dynamic variables which can be defined by the pre-request scripts.

Reply:
noted. re-investigating.